state/state.go

```diff
@@ -40,23 +40,53 @@ type Provider interface {
 }
 
 // Configure the state provider
-func Configure(c *config.Config) (Provider, error) {
-	if len(c.TFE.Token) > 0 {
-		log.Info("Using Terraform Enterprise as the state/locks provider")
-		return NewTFE(c)
+func Configure(c *config.Config) ([]Provider, error) {
+	var providers []Provider
+	if len(c.TFE) > 0 {
+		objs, err := NewTFECollection(c)
+		if err != nil {
+			return []Provider{}, err
```

Comment:
If you named the return values of the method, you might just use `return` here iirc

Reply:
Note, if one provider fails, you get no providers at all at the moment.

Reply:
> Note, if one provider fails, you get no providers at all at the moment.
Do you think it's better to just log an error and continue with the following ones?
Initially, I considered this error as fatal since it would lead to the non-inclusion of an entire provider and because, potentially, it could come from an error in the configuration file which could impact the rest.

Reply:
Not sure. I say I'd prefer to see some data and get an error somewhere rather than nothing at all.

Reply:
> If you named the return values of the method, you might just use return here iirc

Indeed but in this case I think it's more simple like this since if I name the return values I can't use `:=` anymore with err which will force me to declare the temp variable `objs` before each time I use it. It's, imo, more complex and furthermore it raise an error on `gosimple` if I do it.